File: segundo/1er cuatri/PDOO/ejercicios_clase/clase18-11/src/clase18/pkg11/Jugador.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package clase18.pkg11;

/**
 *
 * @author alvarogz
 */
public class Jugador extends Personaje {
    private int nivel;
    
    public Jugador(String nombre, int salud, int nivel){
        super(nombre, salud);
        this.nivel = nivel;
    }
    
    public void subirNivel(){
        this.nivel++;
    }
    
    public void atacar(){
        super.atacar();
        System.out.print( " con el nivel " + nivel);
    }
}

File: segundo/1er cuatri/PDOO/ejercicios_clase/clase1811-ruby/jugador.rb
# frozen_string_literal: true
require_relative 'personaje'

class Jugador < Personaje
  def initialize(nombre, salud, nivel)
    super(nombre, salud)
    @nivel = nivel
  end

  def subir_nivel
    @nivel += 1
  end

  def atacar
    super
    puts " con el nivel #{@nivel}"
  end

end
